Update build to detect if running less than Java 17 and report an error

WildFly Preview now requires Java 17 - as running this feature pack is part of this build Java 17 needs to be used.

We may need to consider if we need other parts of the project to cross compile for Java 11 but for now setting 17 should be sufficient.

Without this the first sign of an issue is an error similar to:


```
[INFO] --- wfgp:7.1.2.Final:build-feature-pack (wildfly-grpc-feature-pack-build) @ wildfly-grpc-preview-feature-pack ---
Forked embedded process has failed with the following error:
java.lang.IllegalStateException: WFLYEMB0022: Cannot invoke 'start' on embedded process
        at org.wildfly.core.embedded.EmbeddedManagedProcessImpl.invokeOnServer(EmbeddedManagedProcessImpl.java:108)
        at org.wildfly.core.embedded.EmbeddedManagedProcessImpl.start(EmbeddedManagedProcessImpl.java:53)
        at org.wildfly.galleon.plugin.featurespec.generator.FeatureSpecGenerator.readFeatureSpecs(FeatureSpecGenerator.java:324)
        at org.wildfly.galleon.plugin.featurespec.generator.FeatureSpecGenerator.forkedForEmbedded(FeatureSpecGenerator.java:240)
        at org.wildfly.galleon.plugin.server.ForkedProcessRunner.main(ForkedProcessRunner.java:56)
Caused by: org.wildfly.core.embedded.EmbeddedProcessStartException: WFLYEMB0021: Cannot start embedded process
        at org.wildfly.embedded@25.0.0.Final//org.wildfly.core.embedded.EmbeddedStandaloneServerFactory$StandaloneServerImpl.start(EmbeddedStandaloneServerFactory.java:295)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.wildfly.core.embedded.EmbeddedManagedProcessImpl.invokeOnServer(EmbeddedManagedProcessImpl.java:96)
        ... 4 more
Caused by: java.util.concurrent.ExecutionException: JBTHR00005: Operation failed
        at org.jboss.threads.AsyncFutureTask.get(AsyncFutureTask.java:253)
        at org.wildfly.embedded@25.0.0.Final//org.wildfly.core.embedded.EmbeddedStandaloneServerFactory$StandaloneServerImpl.start(EmbeddedStandaloneServerFactory.java:278)
        ... 9 more
Caused by: java.lang.Exception: WFLYSRV0056: Server boot has failed in an unrecoverable manner; exiting. See previous messages for details. - Server configuration file in use: standalone.xml - Minimum feature stability level: community
        at org.jboss.as.server@25.0.0.Final//org.jboss.as.server.ServerService.boot(ServerService.java:435)
        at org.jboss.as.controller@25.0.0.Final//org.jboss.as.controller.AbstractControllerService$1.run(AbstractControllerService.java:358)
        at java.base/java.lang.Thread.run(Thread.java:829)
```

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -20,7 +20,7 @@ jobs:
         id: setup-jdk
         uses: actions/setup-java@v4
         with:
-          java-version: 11
+          java-version: 17
           distribution: temurin
           cache: maven
       - name: Install
@@ -30,7 +30,7 @@ jobs:
         id: setup-maven-settings
         uses: actions/setup-java@v4
         with: # running setup-java again overwrites the settings.xml
-          java-version: 11
+          java-version: 17
           distribution: temurin
           server-id: ossrh
           server-username: OSSRH_USERNAME

--- a/.github/workflows/verify.yml
+++ b/.github/workflows/verify.yml
@@ -20,7 +20,7 @@ jobs:
         id: setup-jdk
         uses: actions/setup-java@v4
         with:
-          java-version: '11'
+          java-version: '17'
           distribution: 'temurin'
           cache: maven
       - name: Install
@@ -33,10 +33,10 @@ jobs:
     timeout-minutes: 30
     steps:
       - uses: actions/checkout@v4
-      - name: Set up JDK 11
+      - name: Set up JDK 17
         uses: actions/setup-java@v4
         with:
-          java-version: 11
+          java-version: 17
           distribution: 'temurin'
           cache: 'maven'
       - name: Format Source

--- a/pom.xml
+++ b/pom.xml
@@ -117,7 +117,7 @@
         <galleon.fork.embedded>true</galleon.fork.embedded>
         <galleon.log.time>true</galleon.log.time>
         <galleon.wildfly.home>/tmp</galleon.wildfly.home>
-        <java.version>11</java.version>
+        <java.version>17</java.version>
         <linkXRef>false</linkXRef>
         <maven.compiler.source>${java.version}</maven.compiler.source>
         <maven.compiler.target>${java.version}</maven.compiler.target>

